src/main/java/com/synopsys/integration/detect/configuration/help/yaml/HelpYamlWriter.java

```diff
@@ -0,0 +1,105 @@
+        File helpYaml = new File(buildDir, filename);
+
+        try (FileWriter fileWriter = new FileWriter(helpYaml);
+             BufferedWriter bufferWriter = new BufferedWriter(fileWriter)   ) {
+            writeHeader(bufferWriter);
+            writeProperties(bufferWriter, filteredProperties);
+            logger.info("{} was created at {}", filename, helpYaml.getAbsolutePath());
+        } catch (IOException e) {
+            logger.info("There was an error creating the help yaml file.", e);
+            throw new RuntimeException(e);
+        }
+    }
+    private void writeHeader(BufferedWriter buffer) throws IOException {
+        String bannerString = ResourceUtil.getResourceAsString(this.getClass(), "/banner.txt", StandardCharsets.UTF_8.toString());
+        List<String> bannerStrings = Arrays.asList(bannerString.split("\n"));
+        for (String line : bannerStrings) {
+            buffer.write("# " + line);
+            buffer.newLine();
+        }
+
+        String versionFileString = ResourceUtil.getResourceAsString(this.getClass(), "/version.txt", StandardCharsets.UTF_8.toString());
+        List<String> versionFileContents = Arrays.asList(versionFileString.split("\n"));
+        String versionText = DetectInfoUtility.parseValueFromVersionFileContents(versionFileContents, "version");
+        buffer.write("# v" + versionText + "\n\n");
+
+        String yamlInfoFileString = ResourceUtil.getResourceAsString(this.getClass(), "/help-yaml-header.txt", StandardCharsets.UTF_8.toString());
+        buffer.write(yamlInfoFileString);
+    }
+    private void writeProperties(BufferedWriter buffer, List<Property> properties) throws IOException {
+        List<Property> sortedProperties = properties.stream()
+                .sorted(SORT_BY_GROUP_THEN_KEY)
+                .collect(Collectors.toList());
+
+        String group = null;
+        for (Property p : sortedProperties) {
+            String currentGroup = p.getPropertyGroupInfo().getPrimaryGroup().getName();
+            if (group == null) {
+                group = currentGroup;
+                writeGroup(buffer, group);
+            } else if (!group.equals(currentGroup)) {
+                group = currentGroup;
+                writeGroup(buffer, group);
+            }
+            writeProperty(buffer, p);
+        }
+    }
+    private void writeGroup(BufferedWriter buffer, String groupName) throws IOException {
+        buffer.write(String.format("\n\n##\n# %S\n##", groupName));
+    }
+    private void writeProperty(BufferedWriter buffer, Property property) throws IOException {
+        buffer.write("\n#" + property.getKey() + ": ");
+        if (property.describeDefault() != null) {
+            buffer.write(property.describeDefault());
+        }
+
+        buffer.write("\n\t\t# " + property.getPropertyHelpInfo().getShortText());
+
+        List<String> acceptableValues;
+        if (!(acceptableValues = property.listExampleValues()).isEmpty()) {
```

Comment:
Something that I too do but better not! We should move the assignment to its own line.

Reply:
Done!